src/lib/openjp2/jp2.c

```diff
@@ -2636,6 +2658,104 @@ static OPJ_BOOL opj_jp2_read_ftyp(opj_jp2_t *jp2,
     jp2->jp2_state |= JP2_STATE_FILE_TYPE;
 
     return OPJ_TRUE;
+}
+
+static OPJ_BOOL opj_jp2_read_asoc(	opj_jp2_t *jp2,
+									OPJ_BYTE * p_header_data,
+									OPJ_UINT32 p_header_size,
+									opj_event_mgr_t * p_manager)
+{
+    OPJ_UINT32 label_tag;
+    OPJ_UINT32 asoc_tag;
+    OPJ_UINT32 asoc_size;
+    opj_jp2_asoc_t *asoc;
+
+    /* preconditions */
+    assert(jp2 != 00);
+    assert(p_header_data != 00);
+    assert(p_manager != 00);
+
+    opj_read_bytes(p_header_data,&asoc_size,4);
+    p_header_data += 4;
+    p_header_size -= 4;
+
+    opj_read_bytes(p_header_data,&label_tag,4);
+    p_header_data += 4;
+    p_header_size -= 4;
+    asoc_size -= 4;
+
+    if (label_tag != JP2_LBL) {
+        /* TODO: Verify that ASOC must have a following label ? */
+        opj_event_msg(p_manager, EVT_WARNING, "ASOC data does not have a label (LBL)\n");
+            return OPJ_TRUE; // No error if we could not parse
+    }
+
+    if ( jp2->numasoc == 0 ) {
+        jp2->numasoc = 1;
+        jp2->asoc = opj_malloc(sizeof(opj_jp2_asoc_t));
+    }
+    else {
+        (jp2->numasoc)++;
+        jp2->asoc = opj_realloc(jp2->asoc, jp2->numasoc * sizeof(opj_jp2_asoc_t));
+    }
+    asoc = &(jp2->asoc[jp2->numasoc-1]);
+    asoc->level = jp2->numasoc-1; /* TODO: This is not correct if a parent asoc contains multiple child asocs! */
+    asoc->label_length = asoc_size;
+    asoc->label = opj_malloc(asoc_size);
+    memcpy(asoc->label, p_header_data, asoc_size);
+    asoc->xml_buf = 00;
+    asoc->xml_len = 0;
+
+    p_header_data += asoc_size;
+    p_header_size -= asoc_size;
+
+    opj_read_bytes(p_header_data,&asoc_tag,4);

```

Comment:
ypi sjpimd check that there's at least 4 remaining bytes

Reply:
Fixed